pkg/daemon/spawn_unix.go

```diff
@@ -7,11 +7,15 @@ import (
 	"syscall"
 )
 
-func procAttrForSpawn() *os.ProcAttr {
+func procAttrForSpawn(stdout *os.File) *os.ProcAttr {
+	// The daemon should not inherit a reference to the stdin file descriptor
+	// of the original shell. The daemon shouldn't read anything from stdin so
+	// use the null device.
+	devnull, _ := os.OpenFile("/dev/null", os.O_RDONLY, 0)
```

Comment:
**No. Don't ever do that for stdin/stdout/stderr.** I've debugged two vexing problems in the past due to programs that launched other programs with one or more of those file descriptors closed. UNIX programs read from, or write to, those file descriptors not just in the "main" program but also imported packages (libraries). Every program assumes they are open on something. If they aren't the first `open()` call the program executes will allocate one of those file descriptors. Oops! You now have other parts of the program that use the stdin/out/err file descriptors reading/writing a file not meant to be used as stdin/out/err.

It is extremely unlikely a different path will ever be needed for a Go module predicated on the meta GOOS value "unix".

Reply:
That's a good point, but it seems that it applies more to stdout and stderr, which libraries are likely to write to. I'd be surprised if a library reads from stdin. Nonetheless, it sounds like good practice to avoid closing any of the lowest 3 FDs.